packages/workflow/src/internals.ts

```diff
@@ -629,33 +630,40 @@ export class Activator implements ActivationHandler {
     //
     // Note that there is a deliberately unhandled promise rejection below.
     // These are caught elsewhere and fail the corresponding activation.
-    let input: UpdateInput;
-    try {
-      if (runValidator && this.updateHandlers.get(name)?.validator) {
-        const validate = composeInterceptors(
-          this.interceptors.inbound,
-          'validateUpdate',
-          this.validateUpdateNextHandler.bind(this)
-        );
-        validate(makeInput());
+    const doUpdateImpl = async () => {
+      let input: UpdateInput;
+      try {
+        if (runValidator && this.updateHandlers.get(name)?.validator) {
+          const validate = composeInterceptors(
+            this.interceptors.inbound,
+            'validateUpdate',
+            this.validateUpdateNextHandler.bind(this)
+          );
+          validate(makeInput());
+        }
+        input = makeInput();
+      } catch (error) {
+        this.rejectUpdate(protocolInstanceId, error);
+        return;
       }
-      input = makeInput();
-    } catch (error) {
-      this.rejectUpdate(protocolInstanceId, error);
-      return;
-    }
-    const execute = composeInterceptors(this.interceptors.inbound, 'handleUpdate', this.updateNextHandler.bind(this));
-    this.acceptUpdate(protocolInstanceId);
-    untrackPromise(
-      execute(input)
+      const execute = composeInterceptors(this.interceptors.inbound, 'handleUpdate', this.updateNextHandler.bind(this));
+      this.acceptUpdate(protocolInstanceId);
+      const res = execute(input)
         .then((result) => this.completeUpdate(protocolInstanceId, result))
         .catch((error) => {
           if (error instanceof TemporalFailure) {
             this.rejectUpdate(protocolInstanceId, error);
           } else {
             throw error;
           }
-        })
+        });
+      untrackPromise(res);
+      return res;
+    };
+    untrackPromise(
+      UpdateScope.updateWithInfo(updateId, name, doUpdateImpl).catch((error: any) => {
```

Comment:
That catch statement is useless.

I think in this case, we have sufficient certainty that the promise can't fail. `doUpdateImpl` won't let a throw bubble up. Therefore, you don't even need the `catch()` statement.

Reply:
I was worried about storage.run throwing after calling `storage.disable()` but I think you are right, it does not throw, it just creates another context...

Reply:
TBH, I don't like the fact that we create yet another ASL, due to [a memory issue](https://github.com/temporalio/sdk-typescript/issues/1432). But that's ok for now.